src/integration/control/configureIndex.test.ts

```diff
@@ -0,0 +1,132 @@
+import { BasePineconeError } from '../../errors';
+import { Pinecone } from '../../index';
+import { randomIndexName, waitUntilReady } from '../test-helpers';
+
+describe('configure index', () => {
+  let indexName;
+  let pinecone: Pinecone;
+
+  beforeEach(async () => {
+    pinecone = new Pinecone();
+    indexName = randomIndexName('configureIndex');
+
+    await pinecone.createIndex({
+      name: indexName,
+      dimension: 5,
+      waitUntilReady: true,
+      podType: 'p1.x1',
+      replicas: 2,
+    });
+  });
+
+  afterEach(async () => {
+    await pinecone.deleteIndex(indexName);
+  });
```

Comment:
**question:** Do we need to be creating and deleting an index for each test rather than using `beforeAll` and `afterAll`? It seems like this would lead to slower tests as we have to wait somewhere around 10 seconds for each index to spin up for each test. For this specific suite could we not just create an index once, perform all the tests and operations, and then delete `afterAll` for cleanup?

Reply:
As currently written, this test suite does end up adding quite a bit of time due to the index creation. But for now I want to keep every test fully isolated from others. 

There seems to be some timing issues in this configure endpoint when called multiple times at a row. At one point I had a test here that scaled replicas up and then down again (using the same index) and it was throwing a 500 error at a high rate, even when I waited for status to reach the Ready state. I don't think they envisioned people scaling up and down separated only by milliseconds.